### DISCUSSION TITLE
New Feature: Concept Page

### DISCUSSION THREAD

**Discussion by gwendupre** [cgwire/kitsu#1188] [Ideas]
Hello there!  (Hello @fsiddi )

While making mockups for the new concept section, we came up with two different principles:

- The first concept assumes that assets are already created in Kitsu, and we add concepts linked to these assets.
This principle groups concepts by asset category.
Therefore, an asset and its accessory could be present in both categories (the character and the props).

![concept01_global](https://github.com/cgwire/kitsu/assets/42962633/2b9b0ead-3726-41cd-b498-d2445cabc33e)

![concept01_detail](https://github.com/cgwire/kitsu/assets/42962633/50f04a6f-6635-451d-b58b-35fc1a8aeb1a)

- The second principle is to have concepts added without classification. Each concept would be linked to one or more assets with a "tag" added to link it to the assets present in Kitsu.

![concept02](https://github.com/cgwire/kitsu/assets/42962633/55865a25-8201-4e44-8129-3b1c2c65792a)

![concept02_detail](https://github.com/cgwire/kitsu/assets/42962633/101fc9a2-7286-4d10-a84c-de19492d8275)

Then, independently of these two principles, we are wondering about versioning.

- All concept previews appear side by side on the page with a version number indicator. Each concept version would be an element displayed. This version is either neutral, validated, or rejected. This would create one task per version.

![concept01_single](https://github.com/cgwire/kitsu/assets/42962633/b49a4222-2411-45a1-8a6a-188ed0815930)

- Alternatively, you can display only one preview per concept, and in the comment panel for that concept, you can see the different versions. The status is linked to the concept task rather than to the version.

![concept01_multiple](https://github.com/cgwire/kitsu/assets/42962633/2c1e4d2d-1736-4530-913a-b2c1819d573b)

About the shot, we are wondering if you are doing concept at the shot or sequence level.

Don't hesitate to get back to us!

**Comment by frankrousseau** [comment]
@fsiddi I was wondering if this should include some AI application connections to generate some concepts automatically. I know it's a controversial subject but I think most studios already use it.

**Comment by fsiddi** [comment]
Hi, thank you for exploring this design. My recommendations:

- use the "tagging" of a concept (similar to how shot casting works) - basically the second option you suggest. The idea would be to have a page that shows all concepts and allows for filtering by entity or asset.
- when it comes to versioning:  new versions of the concept should be grouped (second option). In practice an artist will produce a concept, then refine it and address feedback, similar to any other asset. It's fine to have one status for the concept (which is associated to the latest version, similar to other entities)
- about the relation to shots or entities: a concept painting could relate to either.
- when it comes to AI, please keep it as far away as possible from this area. 

I hope this helps!

**Comment by frankrousseau** [comment]
After a discussion with Francesco, we are going to:

- Add a page to drop many concepts, mainly pictures. 
- Each picture will constitute a single concept.
- Each concept will have a task tied to it to get the same conversation scheme as the usual tasks.
- Each concept can be linked to several entities
- Concepts should not be organized but filterable with the following criteria: uploader, entity, and sorted by dates.
- The entity page should show the list of concepts tied to the entity.

**Comment by fsiddi** [comment]
I thought about this further, and I think that adding an additional filter called "entity type", to filter all shots, all characters or props would be useful. This could be sourced from the concept <- entity relation.

**Comment by frankrousseau** [comment]
The development is ongoing. The first draft will soon be available for internal testing.

**Comment by frankrousseau** [comment]
@fsiddi We just shipped a version of the concept page. It still requires some polishing but the features are there. We would like to have your feedback about it. Let us what you think!

**Comment by fsiddi** [comment]
Thanks for the update! I pulled the latest docker image to quickly check, but could not see it. Could you share some instructions/release notes?

**Reply by frankrousseau** [reply]
Sorry for the late reply. There is a new section named Concepts. From there you can upload picture that will be linkable with assets.

**Reply by fsiddi** [reply]
No problem, I pulled the latest version and the feature is there. Great work! 
I'll look for resources to implement filtering, which seems to be the only major thing missing at the moment.

**Reply by frankrousseau** [reply]
Do you mean filtering by asset?

**Reply by fsiddi** [reply]
Yes, or at least I can't see it.

**Comment by lalamax3d** [comment]
@frankrousseau , thanks, i was wondering how to connect (link) a concept with character asset entity.?

**Reply by frankrousseau** [reply]
When you select a concept in the Concepts page. You can link it in the right panel.
